dataprofiler/tests/profilers/profiler_options/test_profiler_options.py

```diff
@@ -503,6 +505,30 @@ def test_eq(self, *mocks):
         options.structured_options.float.precision.sample_ratio = 0.1
         self.assertNotEqual(options, options2)
 
+    def test_json_encode(self, *mocks):
+        option = ProfilerOptions(presets="complete")
+
+        serialized = json.dumps(option, cls=ProfileEncoder)
+
+        expected_class = "ProfilerOptions"
+        expected_options_attributes = {
+            "structured_options",
+            "unstructured_options",
+            "presets",
```

Comment:
This isn't an issue in this PR, but i'm not sure why we save `presets` to self as oppose to just use it to choose them.

Food for thought of something we might want to change. Not sure why'd we serialize it is why I ask.

Reply:
Also we currently don't raise an error if the preset doesn't exist. it just silently continues.

Reply:
`self.assertIn("presets", actual_option_json["data"])` should catch it when `presets` doesn't exist

Reply:
> Also we currently don't raise an error if the preset doesn't exist. it just silently continues.

Not quite sure that is true once everything gets merged in (i.e. `feature/options`). That's something that Liz added when we were adding the new preset. https://github.com/capitalone/DataProfiler/blob/feature/options/dataprofiler/profilers/profiler_options.py#L1593

Reply:
> `self.assertIn("presets", actual_option_json["data"])` should catch it when `presets` doesn't exist

I'm not suggesting we need to make sure it exists, it is a comment that refers to an issue. 

> Not quite sure that is true once everything gets merged in (i.e. feature/options). That's something that Liz added when we were adding the new preset. https://github.com/capitalone/DataProfiler/blob/feature/options/dataprofiler/profilers/profiler_options.py#L1593

Yup, that solves the silent issue